src/test/scala/ch/epfl/bluebrain/nexus/kg/routes/ResourceRoutesSpec.scala

```diff
@@ -541,21 +542,36 @@ class ResourceRoutesSpec
     }
 
     "search for resources on a custom ElasticView" in new View {
-      val query   = Json.obj("query" -> Json.obj("match_all" -> Json.obj()))
-      val result1 = Json.obj("key1"  -> Json.fromString("value1"))
-      val result2 = Json.obj("key2"  -> Json.fromString("value2"))
-      val qr: QueryResults[Json] =
-        UnscoredQueryResults(2L, List(UnscoredQueryResult(result1), UnscoredQueryResult(result2)))
+      val query      = Json.obj("query" -> Json.obj("match_all" -> Json.obj()))
+      val esResponse = jsonContentOf("/view/search-response.json")
+
       when(
-        elastic.search[Json](query,
-                             Set(s"kg_${defaultEsView.name}"),
-                             Uri.Query(Map("size" -> "23", "other" -> "value")))(Pagination(0L, 23)))
-        .thenReturn(Task.pure(qr))
-      Post(s"/v1/views/$account/$project/nxv:defaultElasticIndex/_search?size=23&other=value", query) ~> addCredentials(
+        elastic.searchRaw(mEq(query), mEq(Set(s"kg_${defaultEsView.name}")), mEq(Uri.Query(Map("other" -> "value"))))(
+          any[HttpClient[Task, Json]]()))
+        .thenReturn(Task.pure(esResponse))
+
+      Post(s"/v1/views/$account/$project/nxv:defaultElasticIndex/_search?other=value", query) ~> addCredentials(
         oauthToken) ~> routes ~> check {
         status shouldEqual StatusCodes.OK
-        responseAs[Json] shouldEqual Json.arr(result1, result2)
+        responseAs[Json] shouldEqual esResponse
```

Comment:
can you also add a test for when the body is not a Json, so the parser with return a `Left(Unexpected(...))`